lisa/sut_orchestrator/azure/tools.py

```diff
@@ -81,16 +85,50 @@ def deprovision(self) -> None:
         self.run("-deprovision --force", sudo=True, expected_exit_code=0)
 
     def upgrade_from_source(self) -> None:
-        git = self.node.tools[Git]
-        git.clone(self._src_url, cwd=self.node.working_path)
+        # Pin the WALinuxAgent to v2.9.0.4 because the latest v2.9.1.1
```

Comment:
add a method parameter version for 2.9.0.4, if no version specified, use latest code, otherwise checkout into this version in waagent source code

Reply:
Re-implemented this function based on the comment.

One follow-up question is whether there are any successful test passes with WALinuxAgent v2.9.1.1? In the previous experiment testing Ubuntu with WALinuxAgent v2.9.1.1, it fails to create an IP over IB.